client/api_off_chain_auth.go

```diff
@@ -144,6 +189,133 @@ func (c *Client) RegisterEDDSAPublicKey(spAddress string, spEndpoint string) (st
 	return jsonResult, error1
 }
 
+// RegisterEDDSAPublicKeyV2 - Register EdDSA public key of this client for the given sp address and spEndpoint.
+//
+// To enable EdDSA authentication, you need to config OffChainAuthOptionV2 for the client.
+// The overall register process could be referred to https://docs.bnbchain.org/greenfield-docs/docs/guide/storage-provider/modules/authenticator#workflow.
+//
+// The EdDSA registering process is typically used in a website, e.g. https://dcellar.io,
+// which obtains a user's signature via a wallet and then posts the user's EdDSA public key to a sp.
+//
+// Here we also provide an SDK method to implement this process, because sometimes you might want to test if a given SP provides correct EdDSA authentication or not.
+// It also helps if you want implement it on a non-browser environment.
+//
+// - spEndpoint: The sp endpoint, to which this API will register client's EdDSA public key. It can be found via https://docs.bnbchain.org/greenfield-docs/docs/greenfield-api/storage-providers .
+//
+// - ret1: The register result when invoking SP UpdateUserPublicKey API.
+//
+// - ret2: Return error when registering failed, otherwise return nil.
+func (c *Client) RegisterEDDSAPublicKeyV2(spEndpoint string) (string, error) {
+	appDomain := c.offChainAuthOptionV2.Domain
+	eddsaSeed := c.offChainAuthOptionV2.Seed
+
+	// get the EDDSA private and public key
+	_, userEddsaPublicKey := GetEd25519PrivateKeyAndPublicKey(eddsaSeed)
+	userEddsaPublicKeyStr := hex.EncodeToString(userEddsaPublicKey)
+	log.Info().Msg("userEddsaPublicKeyStr is " + userEddsaPublicKeyStr)
+
+	IssueDate := time.Now().Format(time.RFC3339)
+	// ExpiryDate format := "2023-06-27T06:35:24Z"
+	ExpiryDate := time.Now().Add(time.Hour * 24).Format(time.RFC3339)
+
+	unSignedContent := fmt.Sprintf(unsignedContentTemplateV2, appDomain, c.defaultAccount.GetAddress().String(), userEddsaPublicKeyStr, appDomain, IssueDate, ExpiryDate)
+
+	unSignedContentHash := accounts.TextHash([]byte(unSignedContent))
+	sig, _ := c.defaultAccount.GetKeyManager().Sign(unSignedContentHash)
+	authString := fmt.Sprintf("%s,SignedMsg=%s,Signature=%s", httplib.Gnfd1EthPersonalSign, unSignedContent, hexutil.Encode(sig))
+	authString = strings.ReplaceAll(authString, "\n", "\\n")
+	headers := make(map[string]string)
+	headers["x-gnfd-app-domain"] = appDomain
+	headers["x-gnfd-app-reg-public-key"] = userEddsaPublicKeyStr
+	headers["X-Gnfd-Expiry-Timestamp"] = ExpiryDate
+	headers["authorization"] = authString
+	headers["origin"] = appDomain
+	headers["x-gnfd-user-address"] = c.defaultAccount.GetAddress().String()
+	jsonResult, error1 := httpPostWithHeader(spEndpoint+"/auth/update_key_v2", "{}", headers)
```

Comment:
can be like `return httpPostWithHeader(spEndpoint+"/auth/update_key_v2", "{}", headers)`